miden-tx/src/tests/kernel_tests/test_tx.rs

```diff
@@ -723,7 +755,18 @@ fn test_load_foreign_account_basic() {
     let mock_chain = MockChainBuilder::default().accounts(vec![foreign_account.clone()]).build();
     let advice_inputs = get_mock_advice_inputs(&foreign_account, &mock_chain);
 
-    let tx_context = TransactionContextBuilder::with_standard_account(ONE)
+    // TODO: Temporary fix: Build a native account that has the same code commitment as the foreign
+    // account which is required for this test to pass right now.
+    let native_account_id =
+        AccountId::try_from(ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN).unwrap();
```

Comment:
Is this still applicable? I thought it should be already fixed. cc @Fumuran

Reply:
>we don't currently have a way to load foreign accounts in the TransactionContext which is the problem here

Exactly. The problem is that we can't provide the mast forest of the foreign account's procedures, so we can get the procedure indexes map value, and can't execute the foreign procedure itself. For that purpose we have a `load_account_code()` procedure in the `TransactionExecutor`, but we don't have anything similar for the `TransactionContext` (by the way, should we implement such procedure for it?). 
In the new version of the tests I will use the `TransactionExecutor` instead of `TransactionContext`, so this won't be a problem in future.

Reply:
It sounds easiest to just use the `TransactionExecutor` then in the new tests if it already has the account loading functionality require for this.